portfolio/src/main/java/com/google/sps/servlets/CreatePuzzleServlet.java

```diff
@@ -64,14 +64,18 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
     Puzzle puzzle = new Puzzle();
 
     puzzle.setImageUrl(uploadedFileUrl);
+    puzzle.setDifficulty(difficulty);
 
     PuzzleDao dao = new PuzzleDao();
 
     //Need to double check that puzzle is passed by reference
     dao.create(puzzle);
```

Comment:
`dao.create(puzzle);` returns a reference to the updated object, something like `puzzle = dao.create(puzzle);` should work.